### DISCUSSION TITLE
INMP441 only static

### DISCUSSION THREAD

**Discussion by NonaSuomy** [pschatzmann/arduino-audio-tools#1505] [Q&A] [answered]
Not sure if I just got a bad set of microphones or something is odd with them. Tried 2 so far just static comes across this example https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-webserver/streams-i2s-webserver_wav
with the ESP32-S3. For ESPhome on the esp32 you had to use right channel even when it was pinned out for the left channel. Not sure if same for this code or how to set this. I tested it out on a plain esp32 and got some very quiet whistles. Not sure if I need to swap to Right channel in the code for it to come across louder. But this at the least showed something was getting detected. The S3 is what I was trying to get working with this mic but just massive constant static tried pins 5 6 7 and 7 10 11.
I couldn't seem to find how to change the pinout properly so I hardcode changed the AudioConfig.h
Then I saw a video where someone set config.pin_bck = #; etc for a max amp in the sketch. How do I set this for audio in?

**Comment by pschatzmann** [comment, marked as answer]
There is a whole section for external ADC in the Wiki. 
I don't understand why you are asking these question ?

**Reply by NonaSuomy** [reply]
Thanks, I found the section after looking at ADC. I searched for config.pin which was coming up with nothing. So I configured it like this:

```arduino
#include "AudioTools.h"
AudioWAVServer server("somessid","somepassword"); // Changed to yours
I2SStream i2sStream;    // Access I2S as stream
ConverterFillLeftAndRight<int16_t> filler(LeftIsEmpty); // Fill both channels - or change to RightIsEmpty

void setup(){
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);
  // Start i2s input with default configuration
  Serial.println("starting I2S...");
  auto config = i2sStream.defaultConfig(RX_MODE);
  config.i2s_format = I2S_STD_FORMAT; // If quality is bad change to I2S_LSB_FORMAT https://github.com/pschatzmann/arduino-audio-tools/issues/23
  config.sample_rate = 44100;
  config.channels = 2;
  config.bits_per_sample = 32;
  //config.channel_format = I2S_CHANNEL_FMT_ALL_LEFT;
  config.pin_data = 13; # SD
  config.pin_bck = 32; # SCK
  config.pin_ws = 23; # WS
  i2sStream.begin(config);
  Serial.println("I2S started");
  // Start data sink
  server.begin(i2sStream, config, &filler);
}

// Arduino loop  
void loop() {
  // Handle new connections
  server.copy();  
}
```

All I get is static on the S3. On a plain esp32 I get a really faint whistle. Are my settings wrong?

**Comment by pschatzmann** [comment]
Did you try to remove the converter to figure out where the issue is coming from ?
By the way you messed up with the bit size: 32 bits needs int32_t and not int16_t.

**Reply by NonaSuomy** [reply]
Changed the code to:

```arduino
#include "AudioTools.h"
AudioWAVServer server("somessid","somepassword"); // Changed to yours
I2SStream i2sStream;    // Access I2S as stream
//ConverterFillLeftAndRight<int32_t> filler(LeftIsEmpty); // Fill both channels - or change to RightIsEmpty

void setup(){
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);
  // Start i2s input with default configuration
  Serial.println("starting I2S...");
  auto config = i2sStream.defaultConfig(RX_MODE);
  config.i2s_format = I2S_STD_FORMAT; // If quality is bad change to I2S_LSB_FORMAT https://github.com/pschatzmann/arduino-audio-tools/issues/23
  config.sample_rate = 44100;
  config.channels = 2;
  config.bits_per_sample = 32;
  //config.channel_format = I2S_CHANNEL_FMT_ALL_LEFT;
  config.pin_data = 13; // SD
  config.pin_bck = 32; // SCK
  config.pin_ws = 23; // WS
  i2sStream.begin(config);
  Serial.println("I2S started");
  // Start data sink
  server.begin(i2sStream, config); //, &filler
}

// Arduino loop  
void loop() {
  // Handle new connections
  server.copy();  
}
```
Still no change mic is barely audible on the Plain ESP32 can hear a crackly whistle

**Reply by pschatzmann** [reply]
And you are sure it is not because of the Logger ?

**Reply by NonaSuomy** [reply]
Maybe but now I just get silence on the S3 with serial disabled
```arduino
#include "AudioTools.h"
AudioWAVServer server("somessid","somepassword"); // Changed to yours
I2SStream i2sStream;    // Access I2S as stream
//ConverterFillLeftAndRight<int32_t> filler(LeftIsEmpty); // Fill both channels - or change to RightIsEmpty

void setup(){
  //Serial.begin(115200);
  //AudioLogger::instance().begin(Serial, AudioLogger::Info);
  // Start i2s input with default configuration
  //Serial.println("starting I2S...");
  auto config = i2sStream.defaultConfig(RX_MODE);
  config.i2s_format = I2S_STD_FORMAT; // If quality is bad change to I2S_LSB_FORMAT https://github.com/pschatzmann/arduino-audio-tools/issues/23
  config.sample_rate = 44100;
  config.channels = 2;
  config.bits_per_sample = 32;
  //config.channel_format = I2S_CHANNEL_FMT_ALL_LEFT;
  config.pin_data = 7; // SD
  config.pin_bck = 12; // SCK
  config.pin_ws = 11; // WS
  i2sStream.begin(config);
  //Serial.println("I2S started");
  // Start data sink
  server.begin(i2sStream, config); //, &filler
}
```

**Reply by pschatzmann** [reply]
The code looks good to me, so I guess it is a hardware problem. 
Double check the connections and the pins.

Also [consider this advice](https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working)...

**Comment by NonaSuomy** [comment]
I got a second person to try with their board they know is working with ESPHome and they got the same issue. Any chance you have an ESP32-S3-N16R8 with inmp441 mic to test?

My wires are short 70mm long duponts that I hand crimped so I know they are good tested with a multimeter. Checked and tested different pins with no success on the S3.

Tested on a plain ESP32-Wrover and it had a bit of sound so we know the MIC at the least sort of works.

**Reply by pschatzmann** [reply]
The statement that it sort of works is worrying me: you should get a good signal!
- How does the signal look like if you record a sine tone and send the output to Serial and display it on the Serial Plotter ?
- Try to switch the GND pin: on my ESP32 it is also not working with one of the GND pins.
- Try to power the mic from an external source - your power pin might emit too much noise
- Try with another INMP441 (in case your microphone is broken)
- Try with another S3 
- Try another ESP32 core version: If you are using 2.0.15, try it with 3.0.0 rc1

**Reply by NonaSuomy** [reply]
The weirdness seemed to be cleared up removing the serial from the plain esp32-wrover. The mic works completely fine on it.
esp32-wrover
![6A178F65-BD52-4AFC-9803-473B2ABE1931](https://github.com/pschatzmann/arduino-audio-tools/assets/1906575/3123c71e-8575-4af2-a48d-3f46eda27a50)
![53CF3C23-0146-4B7F-BA63-1B1F7B9F8ABE](https://github.com/pschatzmann/arduino-audio-tools/assets/1906575/3f9f4b1c-9804-4f4e-9a7c-d95a7547aaa7)

the issue is with the esp32-s3 which it doesn’t.

![4CBDAFAA-A172-4EBF-8E05-45F33C894BB2](https://github.com/pschatzmann/arduino-audio-tools/assets/1906575/909c0b4e-2a10-4399-be75-45f0b80273be)


is there a boost mic volume?
I see config.volume but is this for a dac?

**Comment by pschatzmann** [comment]
Try to visualize the signal. The browser might suppress it if it contains too much noise...
You can pump up the volume on your receiving PC...
Apart from that, there is a Wiki about volume management.
